lib/puppet/provider/registry_value/registry.rb

```diff
@@ -174,6 +174,65 @@ def write_value
     end
   end
 
+  BITS_PER_BYTE = 8
+  BYTE_MASK = 0xFF
+
+  def to_byte_array(num, of_length)
+    bytes = []
+    while
+      # mask only least significant byte and prepend
+      bytes.unshift(num & BYTE_MASK)
+
+      # nothing left to shave off
+      break if (num <= BYTE_MASK)
+
+      # shift off the least significant byte and continue
+      num >>= BITS_PER_BYTE
+    end
+
+    pad = of_length - bytes.length
+    bytes.concat([0] * pad) if pad > 0
+
+    bytes
+  end
```

Comment:
ruby has the ability to unpack a string and interpret it as a fixnum of various widths. For example if you have an little endian 32-bit unsigned integer encoded as a string:

```
irb(main):018:0> str = [0xEF, 0xBE, 0xFE, 0xCA].pack('C4')
=> "\xEF\xBE\xFE\xCA"
```

You can do the following to covert to an integer:

```
irb(main):021:0> str.unpack('L4')[0]
=> 3405692655
```

which in hex is

```
irb(main):022:0> str.unpack('L4')[0].to_s(16)
=> "cafebeef"
```

Similarly, a 64-bit unsigned value would be `unpack('Q8')`